### PR TITLE
Use persistent-claim ReadOnly flag when mounting PVs

### LINKED ISSUE #10244: PVClaim Volume Needs to pass ReadOnly bool
The PersistentVolumeClaim volume plugin (persistent_claim.go) needs to pass its ReadOnly flag to the builder being used to create the PV backing the claim.

All PVs are currently mounted ReadWrite.

### PR DESCRIPTION
**Pull request by markturansky**
Resolves #10244 

A valid use case is GCEPD:   Mounted once read/write or mounted many times read-only.

@thockin 

will add unit tests for each volume effected.

### CODE REVIEW

**Comment by zmerlynn** [comment]
WIP still?

**Comment by zmerlynn** [comment]
@k8s-bot retest this please

**Comment by zmerlynn** [comment]
Assigning to @thockin to sequester or review. (Or close if it stays WIP for too much longer.)

**Comment by thockin** [comment]
This is pretty on-the-edge.  I'm inclined to push it to after 1.0, but I'd like a second opinion.  @quinton-hoole - without this, a GCEPD can not be mounted read-only for multiple readers.  I think it can wait until post 1.0

**Comment by zmerlynn** [comment]
Customer hat: I could wait for that fix. It takes a kind of advanced shared-PD use case.

**Comment by markturansky** [comment]
WIP until unit tests, which I just added.  I tend to agree this is an advanced use case.

The volumes have private implementations with private readOnly booleans.   I added an IsReadOnly func to Builder to provide access to that flag.  

Feedback is welcome :)

**Comment by thockin** [comment]
LGTM

@saad-ali since we were talking about read-only today

**Comment by markturansky** [comment]
@saad-ali interested in a review so I can get a LGTM?

**Comment by saad-ali** [comment]
I'll take a look

**Comment by saad-ali** [comment]
@markturansky could you rebase please

**Comment by markturansky** [comment]
@saad-ali rebased

**Comment by saad-ali** [comment]
LGTM

**Comment by brendandburns** [comment]
@k8s-bot test this [testing build queue, sorry for the noise]

**Comment by brendandburns** [comment]
Automatic merge from SubmitQueue

**Comment by derekwaynecarr** [comment]
We need a consistent squash commit policy moving forward.

Having PRs that merge commits where 3 of the 6 are just "rebased and updated" makes our logs really really really hard to follow.

Can we get a formal policy queued up on this?

**Comment by markturansky** [comment]
@derekwaynecarr agreed.  sorry about that.

**Comment by pmorie** [comment]
+1 to squash commit policy; let's discuss this on the next community hangout.

**Comment by thockin** [comment]
We've actually discussed this internally, it's just hard to enforce and
easy to forget.  The policy goes something like this:

Upon merge, all commits should be meaningful units of work.  Use commits to
add clarity to the development and review process.  Before merging, squash
any "fix review feedback", "typo", and "rebased" sorts of commits.  It is
not imperative that every commit in a PR compile and pass tests
independently, but it is worth striving for.  For mass automated fixups
(e.g. automated doc formatting), use one or more commits for the changes to
tooling and a final commit to apply the fixup en masse.  This makes reviews
much easier.

Is it worth writing that down?  As with all rules - enforcement is the key.
